app/models/mosque.rb

```diff
@@ -3,6 +3,13 @@ class Mosque < ApplicationRecord
   has_many :salah_timings, dependent: :destroy
   belongs_to :user
 
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :area_id, presence: true
+  validates :user_id, presence: true
```

Comment:
```suggestion
  validates :name, :address, presence: true
```

The validation for area and user is already added through belongs_to